File: Program_Files/Subprogram.js
// Outer function
function outer() {
    var outerVar = 1;
    // Print out outer variable
    console.log("Outer function: outerVar = " + outerVar);
    // Middle function
    function middle() {
      var middleVar = 2;
      // Print out outer and middle variable
      console.log("Middle function: outerVar = " + outerVar + ", middleVar = " + middleVar);
      // Inner function
      function inner() {
        var innerVar = 3;
        // Print out all 3 variables
        console.log("Inner function: outerVar = " + outerVar + ", middleVar = " + middleVar + ", innerVar = " + innerVar);
      }
      // Call inner function last
      inner();
    }
    // Call middle function second   
    middle();
  }
  
  // Call outermost function first
  outer();
  

File: Program_Files/Subprogram.py
# Online Python-3 Compiler (Interpreter)
def outer_func():
    outer_var = "outer"

    def middle_func():
        middle_var = "middle"

        def inner_func():
            inner_var = "inner"
            print("Inner function")
            print(f"Inner function variable: {inner_var}")
            print(f"Middle function variable: {middle_var}")
            print(f"Outer function variable: {outer_var}")
        
        inner_func()
        print("Middle funciton")
        print(f"Middle function variable: {middle_var}")
        print(f"Outer function variable: {outer_var}")
    
    middle_func()
    print("Outer function")
    print(f"Outer function variable: {outer_var}")

outer_func()